ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/chain/GenesisState.java

```diff
@@ -153,38 +153,58 @@ private static Hash calculateGenesisStateHash(final List<GenesisAccount> genesis
   private static Address parseCoinbase(final GenesisConfigFile genesis) {
     return genesis
         .getCoinbase()
-        .map(Address::fromHexString)
+        .map(str -> withNiceErrorMessage("coinbase", str, Address::fromHexString))
         .orElseGet(() -> Address.wrap(BytesValue.wrap(new byte[Address.SIZE])));
   }
 
+  private static <T> T withNiceErrorMessage(
+      final String name, final String value, final Function<String, T> parser) {
+    try {
+      return parser.apply(value);
+    } catch (final IllegalArgumentException e) {
+      throw createInvalidBlockConfigException(name, value, e);
+    }
+  }
+
+  private static IllegalArgumentException createInvalidBlockConfigException(
+      final String name, final String value, final IllegalArgumentException e) {
+    return new IllegalArgumentException(
+        "Invalid " + name + " in genesis block configuration: " + value, e);
+  }
+
   private static Hash parseParentHash(final GenesisConfigFile genesis) {
-    return Hash.wrap(Bytes32.fromHexString(genesis.getParentHash()));
+    return withNiceErrorMessage("parentHash", genesis.getParentHash(), Hash::fromHexStringLenient);
   }
 
   private static BytesValue parseExtraData(final GenesisConfigFile genesis) {
-    return BytesValue.fromHexString(genesis.getExtraData());
+    return withNiceErrorMessage("extraData", genesis.getExtraData(), BytesValue::fromHexString);
   }
 
   private static UInt256 parseDifficulty(final GenesisConfigFile genesis) {
-    return UInt256.fromHexString(genesis.getDifficulty());
+    return withNiceErrorMessage("difficulty", genesis.getDifficulty(), UInt256::fromHexString);
   }
 
   private static Hash parseMixHash(final GenesisConfigFile genesis) {
-    return Hash.wrap(Bytes32.fromHexString(genesis.getMixHash()));
-  }
-
-  private static long parseNonce(final GenesisConfigFile genesis) {
-    String nonce = genesis.getNonce().toLowerCase(Locale.US);
-    if (nonce.startsWith("0x")) {
-      nonce = nonce.substring(2);
-    }
-    return Long.parseUnsignedLong(nonce, 16);
+    return withNiceErrorMessage("mixHash", genesis.getMixHash(), Hash::fromHexStringLenient);
```

Comment:
I assume you meant to make the mixHash parsing more lenient here^ ?  Wonder if we should accept any zero-valued hash, and otherwise be strict ...

Reply:
Yes this was deliberate (as with parentHash).  It's unlikely that it will make any difference but since the mix hash for the genesis file really doesn't matter (as long as it's the same for all nodes) it seems nice to not have to add all the digits for no real reason.  It doesn't seem worth the extra code to specifically allow it only for 0x0.